hal/src/hal-sdcard.adb

```diff
@@ -0,0 +1,382 @@
+with Ada.Text_IO; use Ada.Text_IO;
```

Comment:
Why is the driver layer relying on the protocol layer?
How different is SDMMC from I2C or SPI? (Sending command, receiving/sending data)